packages/taquito-michelson-encoder/src/tokens/map.ts

```diff
@@ -107,6 +107,13 @@ export class MapToken extends Token {
       });
   }
 
+  public TypecheckValue(val: unknown) {
+    const err = this.isValid(val);
```

Comment:
As this case is quite complex, I think we should have proper testing:
- a test where we retrieve the storage of a contract having a map inside a map